app/controllers/contactPreference/EmailPreferenceController.scala

```diff
@@ -54,8 +61,24 @@ class EmailPreferenceController @Inject()(errorHandler: ErrorHandler,
       formYesNo.bindFromRequest().fold (
         formWithErrors => Future.successful(BadRequest(emailPreferenceView(formWithErrors))),
         {
-          case Yes => Future.successful(Redirect(controllers.contactPreference.routes.EmailToUseController.show())
-            .addingToSession(SessionKeys.contactPrefUpdate -> "true"))
+          case Yes =>
+            vatSubscriptionService.getCustomerInfo(user.vrn).map {
+              case Right(details) =>
+               details.ppob.contactDetails.flatMap(_.emailAddress) match {
+                  case Some(_) =>
+                    Redirect(controllers.contactPreference.routes.EmailToUseController.show())
+                      .addingToSession(SessionKeys.contactPrefUpdate -> "true")
+                  case None =>
+                    Redirect(controllers.contactPreference.routes.AddEmailAddressController.show())
+                      .addingToSession(SessionKeys.contactPrefUpdate -> "true")
+
+                }
```

Comment:
You could do something like
```suggestion
val result = details.ppob.contactDetails.flatMap(_.emailAddress) match {
  case Some(_) =>
    Redirect(controllers.contactPreference.routes.EmailToUseController.show())
  case None =>
    Redirect(controllers.contactPreference.routes.AddEmailAddressController.show())
}
result.addingToSession(SessionKeys.contactPrefUpdate -> "true")
```
so you don't have to write the adding to session bit twice